src/github.com/Virepri/Shoraldele/Buffer/buffer.go

```diff
@@ -34,3 +36,43 @@ func GetBufferContents(offset int, length int) []byte {
 func GetBufferSize() int {
 	return len(work_buffer)
 }
+
+func GetCursorPosition() uint64 {
+	return cursorPosition
+}
+
+func SetCursorPosition(newpos uint64) {
+	cursorPosition = newpos
+	//TODO: Call some function to notify output module
+}
+
+//Wraps at 80 chars, hardcoded 'cause codeday
+//Each element is [length, offset]
+func BufferLineSplit() (res [][2]int) {
```

Comment:
Creative; not the way I would have done it. But this keeps the duplication down. I like it.